src/atomate2/vasp/flows/defect.py

```diff
@@ -0,0 +1,224 @@
+            struct1,
+            struct2,
+            distortions=self.distortions,
+            static_maker=self.static_maker,
+            prev_vasp_dir=dir1,
+            add_name="q1",
+        )
+
+        deformations2 = spawn_energy_curve_calcs(
+            struct2,
+            struct1,
+            distortions=self.distortions,
+            static_maker=self.static_maker,
+            prev_vasp_dir=dir2,
+            add_name="q2",
+        )
+
+        deformations1.append_name(" q1")
+        deformations2.append_name(" q2")
+
+        ccd_job = get_ccd_from_task_docs(
+            deformations1.output, deformations2.output, struct1, struct2
+        )
+
+        return Flow(
+            jobs=[
+                charged_structures,
+                relax1,
+                relax2,
+                deformations1,
+                deformations2,
+                ccd_job,
+            ],
+            output=ccd_job.output,
+            name=name,
+        )
+
+
+@job
+def get_charged_structures(structure: Structure, charges: Iterable):
+    """Adding charges to structure.
+
+    This needs to be a job so the results of other jobs can be passed in.
+
+    Parameters
+    ----------
+    structure
+        A structure.
+    charges
+        A list of charges on the structure
+
+    Returns
+    -------
+    dict
+        A dictionary with the two structures with the charge states added.
+
+    """
+    structs_out = [structure.copy() for _ in charges]
+    for i, q in enumerate(charges):
+        structs_out[i].set_charge(q)
+    return structs_out
+
+
+@dataclass
+class NonRadiativeMaker(Maker):
+    """Class to generate workflows for the calculation of the non-radiative defect capture."""
+
+    ccd_maker: ConfigurationCoordinateMaker
+    name: str = "non-radiative"
+    wswq_maker: FiniteDifferenceMaker = field(
+        default_factory=lambda: FiniteDifferenceMaker()
+    )
+
+    def make(
+        self,
+        structure: Structure,
+        charge_state1: int,
+        charge_state2: int,
+    ):
+        """Create the job for Non-Radiative defect capture.
+
+        Make a job for the calculation of the configuration coordinate diagram.
+        Also calculate the el-phon matrix elements for 1-D special phonon.
+
+        Parameters
+        ----------
+        structure
+            A structure.
+        charge_state1
+            The reference charge state of the defect.
+        charge_state2
+            The excited charge state of the defect
+
+        """
+        name = f"{self.name}: {structure.formula}({charge_state1}-{charge_state2})"
+        flow = self.ccd_maker.make(
+            structure=structure,
+            charge_state1=charge_state1,
+            charge_state2=charge_state2,
+        )
+        ccd: CCDDocument = flow.output
+
+        dirs0 = ccd.distorted_calcs_dirs[0]
+        dirs1 = ccd.distorted_calcs_dirs[1]
+        mid_index0 = len(self.ccd_maker.distortions) // 2
```

Comment:
I think it could be nice to store the undistorted index in the CCDDocument itself. Might be useful later on and then you could use it directly here?